src/services/appointmentData.service.js

```diff
@@ -54,28 +66,71 @@ export function sortData(data, sortKey) {
             return first - second;
         }
     });
-    return newData;
 }
 
 export function filterData(data, { filterByAvailable, filterByZipCode }) {
     return data.filter((d) => {
         if (filterByAvailable && !isAvailable(d)) {
             return false;
         }
-        if (
+        return !(
             filterByZipCode.zipCode &&
             !isWithinRadius(d, filterByZipCode.zipCode, filterByZipCode.miles)
-        ) {
-            return false;
-        }
-        return true;
+        );
     });
 }
 
 export function getAppointmentData() {
+    let testDataTransformed = null;
+
+    if (process.env.NODE_ENV === "development") {
+        // This is a testing branch to get data from a local file instead of the production file.
+        // It will read a file called "test/devtest.json" in the src directory.
+        // You can obtain a cached file using a cmd line:
+        //
+        // aws s3 cp s3://ma-covid-vaccine/data-2021-02-23T2253Z.json src/test/devtest.json
+
+        try {
+            let testData = require("../test/devtest.json");
+
+            // If it has 'body', then this looks like something pasted from a browser (View Source)
+            if (testData.hasOwnProperty("body")) {
+                if (typeof testData.body == "object") {
+                    testData = testData.body;
+                } else {
+                    testData = JSON.parse(testData.body);
+                }
+            }
+
+            if (testData.results) {
+                if (testData.timestamp) {
+                    dataNow = dayjs.utc(
+                        testData.timestamp,
+                        "YYYY-MM-DDTHHmmss[Z]"
+                    );
+                } else {
+                    // If there is no timestamp in the test data
+                    // then just use a date in the past so that nothing is filtered
+                    dataNow = dayjs("03/11/2020");
+                }
+                testDataTransformed = transformData(testData.results);
+            }
+        } catch (err) {
+            // if the file doesn't exist, just get the prod file
+            testDataTransformed = null;
+        }
+    }
+
+    // If there is not any test d
+    if (!testDataTransformed) {
```

Comment:
Will this use our endpoint if there's no test data? That would be ideal!

(I'd test but I'm trying to do this from my phone while I feed the baby)

Reply:
Exactly!  If you want to run the code with _live data_ then you just have to delete the "test/devtest.json" file (or have a malformed one)